add-scroll-to-top-feature(#54)

Asheklm6 opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'.  -->
<!-- If there is no issue being resolved, please open one before creating this pull request. -->
<!-- Replace [issue number] with the issue number (don't leave the square brackets)--likewise for [issue author]. -->
- Fixes #54 by @possumbilities

## Description
<!-- Concisely describe what the pull request does. -->
This PR implements a scroll-to-top button that becomes visible after the user scrolls down a certain distance on the page, improving navigation for users.

## Technical details
<!-- Add any other information or technical details about the implementation; or delete this section entirely. -->

Added a scroll-to-top button in index.html.
Styled the button using CSS in style.css to ensure it appears fixed at the bottom right of the screen.
The button is initially hidden and designed to appear after scrolling down.

## Tests
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or comment out this section entirely. -->
Open index.html in a browser.
Scroll down the page to verify that the button appears after scrolling down

## Screenshots
<!-- Add screenshots to show the problem and the solution; or comment out this section entirely. -->
Scroll before.
![Screenshot (111)](https://github.com/user-attachments/assets/2b583756-4a02-44c1-906d-f120b33cc5f0)

Scroll After

![Screenshot (112)](https://github.com/user-attachments/assets/6fb86ad2-e5ca-41e4-8659-6f4b2737992f)



## Checklist
<!-- DON'T remove this section or any of the lines. -->
<!-- Leave incomplete or inapplicable lines unchecked. -->
<!-- Replace the [ ] with [x] to check the boxes (there is no space between x and square brackets). -->
- [x ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ x] My commit messages follow [best practices][best_practices].
- [x ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ x] I added or updated documentation (if applicable).
- [ x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- YOU MUST READ AND UNDERSTAND THE FOLLOWING ATTESTATION. -->
<!-- -->
<!-- Be aware that copying and pasting from discussion sites or generative AI isn't allowed under this DCO. -->

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

Asheklm6 committed:
add-scroll-to-top-feature

SisiVero commented:
hi @Asheklm6 , well done. 
where you have "Fixes #[54] by @[asheklm6]", remove the "[" and "]" . just write #54 so as to link your PR to the issue #54 that you're about to fix.
Also instead of your username "@[asheklm6]" write @possumbilities as she is the one that opened the issue.
You're doing great. Well done.

Asheklm6 commented:
@SisiVero I've made the changes as you suggested. I've updated the PR description to remove the brackets around the issue number and replaced my username with @possumbilities. Thank you for your guidance . Please let me know if there's anything else I should adjust.

SisiVero commented:
@Asheklm6 , well done.
Your PR details are okay now.
@possumbilities is in charge of reviewing PRs and will have a look at it by Monday.

Asheklm6 committed:
Add custom header rules

possumbilities commented on _headers:
```diff
@@ -0,0 +1,12 @@
+
+/images/*           # Applies to all files in the /images/ folder
+  Cache-Control: max-age=31536000
+  Access-Control-Allow-Origin: *
```
Changes here are unrelated to this PR

possumbilities commented on docs/index.html:
```diff
@@ -1025,5 +1025,13 @@ <h1>Name</h1>
 document.querySelectorAll("code.lang-html").forEach(el => el.innerText = el.innerHTML);
 </script>
 
+
```
extra space added.

possumbilities commented on docs/style.css:
```diff
@@ -26,4 +30,58 @@ span.distinct {
+
+
+
+   
```
missing EOF character

possumbilities commented:
Hi @Asheklm6 thank you for working on this! 

The feature in question is for the Vocabulary UX, not the docs. You've added this element to the docs instead. 

Your solution here would need to be added to the Vocabulary contexts themselves, not the documentation.
